Auto center spacemacs-buffer on window resize.

ralesi opened the pull request:
Addresses some of the concerns in #5120.  Will resize spacemacs-buffer in current frame if any window is resized.

bmag commented on core/core-spacemacs-buffer.el:
```diff
@@ -811,18 +811,22 @@ border."
       (page-break-lines-mode)
       (save-excursion
+        (when refresh
+          (let ((buffer-read-only nil))
```
`inhibit-read-only` was made for this

ralesi replied:
How is one better than another for single command usage?

bmag replied:
`inhibit-read-only` also ignores read-only text properties (see the variable's documentation)

bmag commented on core/core-spacemacs-buffer.el:
```diff
@@ -841,12 +845,20 @@ already exist, and switch to it."
+    (when (and space-win
+               (not (window-minibuffer-p frame-win)))
+      (save-selected-window
+        (select-window space-win)
```
`with-selected-window` is more concise

ralesi replied:
I agree, I tried that first and it didn't work, but I think that was likely because I was still killing the spacemacs-buffer instead of erasing it.

syl20bnr replied:
I tried with `with-selected-window` and it works.

syl20bnr commented:
That's really great 💜 
Do you think it is also possible to resize when the actual windows manager's window (frame) is resized ? For instance in i3 it does not refresh the buffer when I resize the X window.

ralesi commented:
@syl20bnr, I found bugs in different builds of Emacs 25.X that would allow/disallow it to work with frame resizing.  The latest build of 25. seemed to work for windows when a build from back in March did not trigger on frame resize.  Can you try the latest build of Emacs to see if it works?    Latest changes suggested have addressed the code suggestions.

ralesi commented:
@syl20bnr, I updated with a few fixes, and it now uses `window-configuration-change-hook`. So should work everywhere.  Also checks for change in buffer width and only fires on change.

deb0ch commented:
I tried it out and when homepage banner is the random ascii ones, everytime you resize it will load a new different one. It does not break functionnality, but is just a little bit disconcerting.

ralesi commented:
I also notice with the random banner that the ASCII is not centered, will have to look into how to do that

syl20bnr commented:
@ralesi not needed there is already a PR for this :-)

ralesi commented:
@syl20bnr,  which PR is that?

d12frosted commented:
@ralesi here it is: #6502 💃

TheBB commented:
Thanks! Cherry-picked in develop. You can safely delete your branch.

TheBB commented:
Btw, I guess you didn't try this commit with `dotspacemacs-startup-banner` set to `random`. :joy:

https://github.com/syl20bnr/spacemacs/commit/df255da69a180c25c9fce5d7db7618799af284f7
